main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2356,17 +2457,75 @@ object Defaults extends BuildCommon {
     }
     analysis
   }
-  def compileIncrementalTask = Def.task {
-    val s = streams.value
-    val ci = (compile / compileInputs).value
-    val ping = earlyOutputPing.value
-    val reporter = (compile / bspReporter).value
-    BspCompileTask.compute(bspTargetIdentifier.value, thisProjectRef.value, configuration.value) {
-      task =>
-        // TODO - Should readAnalysis + saveAnalysis be scoped by the compile task too?
-        compileIncrementalTaskImpl(task, s, ci, ping, reporter)
-    }
-  }
+
+  def compileIncrementalTaskSettings(key: TaskKey[(Boolean, HashedVirtualFileRef)]) =
+    inTask(key)(
+      Seq(
+        (TaskZero / key) := (Def
```

Comment:
> What does this pattern do?

So `inTask(key)(Seq(...))` is useful when you want to task-scope a bunch of setting references to some `key`, and reuse it. The best example is `packageTaskSettings(...)` where it's used to define `mappings` etc scoped to `packageBin` etc like:

```scala
      packageTaskSettings(packageBin, packageBinMappings) ++
      packageTaskSettings(packageSrc, packageSrcMappings) ++
      packageTaskSettings(packageDoc, packageDocMappings) ++
```

Now if you want to define `packageBin` itself in `packageTaskSettings`, you don't want to task-scope `key`, so you write `TaskZero / key` so you get an undecorated `packageBin` task.

> Also I suggest we inline `compileInternal` here instead of `key`. It would make things more explicit.

Yea, I think that makes sense since there's no reuse.

Reply:
Fixed in 390c7bbc82f94e742b4b8d5d5a9d3fa44c636880

Reply:
> Now if you want to define packageBin itself in packageTaskSettings, you don't want to task-scope key, so you write TaskZero / key so you get an undecorated packageBin task.

Would it be clearer to write it that way?
```scala
Def.settings(
  key := ???,
  inTask(key)(
     ... // other task-scoped keys
  )
)
```